tests/qos/test_qos_sai.py

```diff
@@ -183,11 +187,23 @@ def testQosSaiHeadroomPoolSize(self, ptfhost, dutTestParams, dutConfig, dutQosCo
             "dst_port_id": qosConfig["hdrm_pool_size"]["dst_port_id"],
             "dst_port_ip": testPortIps[qosConfig["hdrm_pool_size"]["dst_port_id"]],
             "pgs_num": qosConfig["hdrm_pool_size"]["pgs_num"],
-            "pkts_num_leak_out": qosConfig["pkts_num_leak_out"],
             "pkts_num_trig_pfc": qosConfig["hdrm_pool_size"]["pkts_num_trig_pfc"],
+            "pkts_num_leak_out": qosConfig["pkts_num_leak_out"],
             "pkts_num_hdrm_full": qosConfig["hdrm_pool_size"]["pkts_num_hdrm_full"],
             "pkts_num_hdrm_partial": qosConfig["hdrm_pool_size"]["pkts_num_hdrm_partial"],
         }
+        pkts_num_trig_pfc_shp = qosConfig["hdrm_pool_size"].get("pkts_num_trig_pfc_shp")
+        if pkts_num_trig_pfc_shp:
+            testParams["pkts_num_trig_pfc_shp"] = pkts_num_trig_pfc_shp
+
+        packet_size = qosConfig["hdrm_pool_size"].get("packet_size")
+        if packet_size:
+            testParams["packet_size"] = packet_size
+            testParams["cell_size"] = qosConfig["hdrm_pool_size"]["cell_size"]
+
+        margin = qosConfig["hdrm_pool_size"].get("margin")
+        if margin:
+            testParams["margin"] = margin
         testParams.update(dutTestParams["basicParams"])
```

Comment:
please do a rebase. There has been some changes in this file lately

Reply:
Done.